framework/wazuh/syscheck.py

```diff
@@ -84,9 +84,13 @@ def last_scan(agent_id):
     :return: Dictionary: end, start.
     """
     my_agent = Agent(agent_id)
-    agent_version = my_agent.get_basic_information(select={'fields': ['version']})['version']
+    # if agent status is never connected, a KeyError happens
+    try:
+        agent_version = my_agent.get_basic_information(select={'fields': ['version']})['version']
+    except KeyError:
+        agent_version = 'ND'
```

Comment:
I think the function can directly return at this point:
```python
return {'start': 'ND', 'end': 'ND'}
```

The applied method used is doing unnecessary function calls. The agent doesn't have any `last scan` if it's _never connected_ so it's not necessary to actually do a database request.